src/crdt.rs

```diff
@@ -1173,4 +1206,29 @@ mod tests {
             assert_eq!(blob.get_id().unwrap(), id);
         }
     }
+    /// FIXME: This is obviously the wrong way to do this. Need to implement leader selection,
+    /// delete this test after leader selection is correctly implemented
+    #[test]
+    fn test_update_leader() {
+        logger::setup();
+        let me = ReplicatedData::new_leader(&"127.0.0.1:1234".parse().unwrap());
+        let lead = ReplicatedData::new_leader(&"127.0.0.1:1234".parse().unwrap());
+        let lead2 = ReplicatedData::new_leader(&"127.0.0.1:1234".parse().unwrap());
+        let mut crdt = Crdt::new(me.clone());
+        assert_matches!(crdt.top_leader(), None);
```

Comment:
`assert_eq`

Reply:
I don't think that will type check, it will complain that None is untyped I think.